index.js

```diff
@@ -0,0 +1,434 @@
+exports = module.exports = lifecycle
+exports.makeEnv = makeEnv
+exports._incorrectWorkingDirectory = _incorrectWorkingDirectory
+
+const spawn = require('./lib/spawn')
+const path = require('path')
+const Stream = require('stream').Stream
+const fs = require('graceful-fs')
+const chain = require('slide').chain
+const uidNumber = require('uid-number')
+const umask = require('umask')
+const which = require('which')
+
+let PATH = 'PATH'
+
+// windows calls it's path 'Path' usually, but this is not guaranteed.
+if (process.platform === 'win32') {
+  PATH = 'Path'
+  Object.keys(process.env).forEach(function (e) {
+    if (e.match(/^PATH$/i)) {
+      PATH = e
+    }
+  })
+}
+
+function logid (pkg, stage) {
+  return pkg._id + '~' + stage + ':'
+}
+
+function lifecycle (pkg, stage, wd, opts) {
```

Comment:
Did a bit of digging, this is rarely equivalent to `process.cwd`, it can be: `pkg.path`, `pkg.fromPath`, `npm.prefix`, `moduleStagingPath`, etc. I'd like to keep it as a formal parameter for now, because even in `cipm` it won't be `process.cwd`.